### PR TITLE
warn for unset variable only during interpolation

### PR DESCRIPTION
**Pull request by ndeloof**
close docker/compose-cli#1918

### CODE REVIEW

**Review by Ocramius — APPROVED** [review]
Overall looks OK to me code-wise, but is there a meaningful way to test this change?

File: template/template.go [hunk 1]
```diff
@@ -97,7 +99,10 @@ func SubstituteWith(template string, mapping Mapping, pattern *regexp.Regexp, su
 			return value
 		}
 
-		value, _ := mapping(substitution)
+		value, ok := mapping(substitution)
+		if !ok {
+			logrus.Warnf("The %s variable is not set. Defaulting to a blank string.", substitution)
```
Comment (ulyssessouza) [inline template/template.go:104]:
```suggestion
			logrus.Warnf("The %q variable is not set. Defaulting to a blank string.", substitution)
```

**Comment by ndeloof** [comment]
just noticed I already pushed this one as https://github.com/compose-spec/compose-go/pull/157

**Review by ulyssessouza — APPROVED** [review]
LGTM

**Pull request merged**